README.md

```diff
@@ -83,10 +83,39 @@ require 'qbwc'
 QBWC.add_job(:list_customers, false, '', CustomerTestWorker)
 ```
 
-After adding a job, it will remain active and will run every time QuickBooks Web Connector runs an update. If you don't want this to happen, you can have custom logic in your worker's `should_run?` or have your job disable or delete itself after completion. 
+After adding a job, it will remain active and will run every time QuickBooks Web Connector runs an update. If you don't want this to happen, you can have custom logic in your worker's `should_run?` or have your job disable or delete itself after completion. For example:
+
+```ruby
+	def handle_response(r, job, data)
+		QBWC.delete_job(job.name)
+	end
+
+```
+
 
 Use the [Onscreen Reference for Intuit Software Development Kits](https://developer-static.intuit.com/qbSDK-current/Common/newOSR/index.html) (use Format: qbXML) to see request and response formats to use in your jobs. Use underscored, lowercased versions of all tags (e.g. `customer_query_rq`, not `CustomerQueryRq`).
 
+### Requests passed to add_job ###
+
+You can optionally provide requests directly to QBWC.add_job. If provided, these requests can return a single qbxml request, or an array of qbxml requests.
+
+The worker requests method overrides requests passed to add_job; if the worker requests method returns a non-nil value, then this value will be used to create the request (and the requests passed to add_job will be ignored).
+
+### Data passed to add_job ###
+
+You can optionally provide arbitrary data directly to QBWC.add_job. If provided, this data will be passed to handle_response.
+
+### Sessions ###
+
+You may optionally specify an initialization block that will be called when each QuickBooks Web Connector session is established:
+
+```ruby
+	QBWC.set_session_initializer() do
+          puts "New QuickBooks Web Connector session has been established"
+        end
```

Comment:
- Would you put this in your job, or what?
- When is the session established? One per job, once per request, once every time you click on the button in web connector?
- Give an example why you would want to do this.